src/pkg/packager/prune.go

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package packager contains operations for working with helm charts.
+package packager
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/zarf-dev/zarf/src/internal/packager/helm"
+	"github.com/zarf-dev/zarf/src/pkg/cluster"
+	"github.com/zarf-dev/zarf/src/pkg/state"
+)
+
+// PruneOptions are the options for Prune
+type PruneOptions struct {
+	Cluster           *cluster.Cluster
+	Component         string
+	Chart             string
+	Timeout           time.Duration
+	NamespaceOverride string
+	Pending           bool
+}
+
+// PruneStateResult is the result of GetPruneableCharts
+type PruneStateResult struct {
+	PruneableCharts map[string][]state.InstalledChart
+}
+
+// GetPruneableCharts returns a list of installed charts that can be pruned per component.
+// This function serves as a coordination layer that delegates the filtering logic to the
+// DeployedPackage and wraps the result in PruneStateResult for the packager layer.
+func GetPruneableCharts(deployedPackage *state.DeployedPackage, opts PruneOptions) (PruneStateResult, error) {
```

Comment:
IMO, we should delete this function. I think it's reasonable to expect SDK users to call functions in the sdk outside of packager such as `deployedPackage.GetPrunableCharts()`